app-backend/tile/src/main/scala/routes/HealthCheckRoute.scala

```diff
@@ -47,19 +43,37 @@ object HealthCheckRoute extends LazyLogging {
     * While the key won't actually be present in the cache, if the cache is reachable,
     * the return from the try will be a Success(null), while if the cache is unreachable,
     * the Try will return a Failure
-    *
-    * memcachedClient.get doesn't return a native Future, but some fancy memcached
-    * Future type that doesn't play well with native futures and makes me sad.
     */
-  def checkCacheHealth: Boolean = {
+  def checkCacheReadHealth: Option[String] = {
     val cacheKey = Random.nextString(8)
     Try { memcachedClient.get(cacheKey) } match {
-      case Success(_) => true
+      case Success(_) => None
       case Failure(_) =>
         logger.error(
           s"Failed reading from memcached client for key $cacheKey"
         )
-        false
+        Some("Failed reading from memcached client")
+    }
+  }
+
+  /** Attempt to write a random key to the cache
+    *
+    * If the write fails, report and return the message that
+    * memcached sent back about the failure
+    */
+  def checkCacheWriteHealth: Option[String] = {
+    val cacheKey = Random.nextString(8)
```

Comment:
I think we should maybe re-use the cache key for the writes so that we don't fill up the cache with random keys

Reply:
Oh I guess it has a short TTL, nevermind